libraries/DataFlash/DataFlash_MAVLinkLogTransfer.cpp

```diff
@@ -155,13 +173,17 @@ void DataFlash_Class::handle_log_request_end(GCS_MAVLINK &link, mavlink_message_
     mavlink_msg_log_request_end_decode(msg, &packet);
```

Comment:
Just an idea: should we verify that the end request comes from the channel we are sending to? It disables a GCS to stop another doing log downloads, but in the other hand, if the downloading GCS crashes, the vehicle will get stuck at log downloading (which isn't a big issue since it would be disarmed and so should be easy to restart).